### DISCUSSION TITLE
DITA codeblock language when converting to Markdown

### DISCUSSION THREAD

**Discussion by kirkilj** [dita-ot/dita-ot#4504] [Q&A]
Using the markdown output transtype, I noticed that the language of a DITA codeblock using the outputclass="language-xxx" syntax in Oxygen is treated as any other outputclass, meaning it gets converted to a Markdown attribute list:
```
``` {.language-yaml}
```
Most of the most common Markdown flavors just put the language name after the tick marks:
```
```yaml
```

Although using attribute lists is consistent from an outputclass treatment perspective, it doesn't make for the most compatible treatment by most Markdown flavors and editors such as VS-Code and other code editors. (I asked for attribute-list support, so I realize I'm going against a prior desire for this one case.)

Would it make sense in LwDITA to special-case the conversion of outputclass for codeblocks with a language prefix in the outputclass so it's compatible with most Markdown flavors?

Our LLM ingester prefers Markdown, so it could help with the representation of code if we can pass along the language in a form most Markdown parsers would recognize. We have to downsample to Markdown for compatibility reasons.

**Comment by raducoravu** [comment]
I see @jelovirt recently worked on this PR: https://github.com/jelovirt/org.lwdita/pull/228
but it seems more to attempt to generate "outputclass="language-X"" values when converting MD to DITA XML, so I'm not sure if it also handles the DITA to markdown generation.

**Reply by kirkilj** [reply]
I saw the same thing and wasn't sure it addressed what I raised in the Discussion section. 

For example the code diff mentions a language prefix option,but I couldn't tell if setting it to null somehow would result in just the language being appended to the back ticks in the DITA to Markdown direction.
